lib/validate_url.rb

```diff
@@ -13,28 +13,32 @@ def initialize(options)
         options.reverse_merge!(message: :url)
         options.reverse_merge!(no_local: false)
         options.reverse_merge!(public_suffix: false)
+        options.reverse_merge!(accept_array: false)
 
         super(options)
       end
 
       def validate_each(record, attribute, value)
         schemes = [*options.fetch(:schemes)].map(&:to_s)
-        begin
-          uri = URI.parse(value)
-          host = uri && uri.host
-          scheme = uri && uri.scheme
+        if value.respond_to?(:each)
+          # Error out if we're not allowing arrays
+          if !options.include?(:accept_array) || !options.fetch(:accept_array)
+            record.errors.add(attribute, :url, **filtered_options(value))
+          end
 
-          valid_raw_url = scheme && value =~ /\A#{URI::regexp([scheme])}\z/
-          valid_scheme = host && scheme && schemes.include?(scheme)
-          valid_no_local = !options.fetch(:no_local) || (host && host.include?('.'))
-          valid_suffix = !options.fetch(:public_suffix) || (host && PublicSuffix.valid?(host, :default_rule => nil))
+          # We have to manually handle `:allow_nil` and `:allow_blank` since it's not caught by
+          # ActiveRecord's own validators. We do that by just removing all the nil's if we want to
+          # allow them so it's not passed on later.
+          value = value.reject(&:nil?) if options.include?(:allow_nil) && options.fetch(:allow_nil)
+          value = value.reject(&:blank?) if options.include?(:allow_blank) && options.fetch(:allow_blank)
 
-          unless valid_raw_url && valid_scheme && valid_no_local && valid_suffix
-            record.errors.add(attribute, options.fetch(:message), value: value)
-          end
-        rescue URI::InvalidURIError
-          record.errors.add(attribute, :url, **filtered_options(value))
+          result = value.map { |v| validate_url(record, attribute, v, schemes) }
+          errors = result.flatten.reject(&:nil?)
+
+          return errors.length.positive? ? errors.first : true
```

Comment:
Seems I'm an old man and I prefer to have `errors.length>0` to explicitly see that there's anything in an array. Also why do you return true if there's no errors?

Reply:
I used `.positive` mostly because that's the Rubocop standard, which I tend to default to when no other style guide is indicated. As for `true` it's because it worked, but I'll change it to `nil`.

Reply:
ok, can we agree on `errors.any? ? errors.first : true`. Just errors.length.positive? doesn't make any sense to me in this case. Rubycop just wishes you use methods instead of manual logic to make code more readable.

Reply:
Absolutely, I'll make the fix up right away.

Reply:
I just pushed a change to fix this, it's a good call for sure.